### PR TITLE
CU-33t2rv1: Play encoded audio file

### PR DESCRIPTION
**Pull request by dmkirshon**
[Encoded_Audio_Demo.webm](https://user-images.githubusercontent.com/86050631/223442118-1f06f443-76f2-422b-841d-6001d0bb58e6.webm)

Includes:
 - Audio playback for song stream url on `Library` page with play/stop ability
   - Icon changes to play on song end
   - Audio playback stops on page change
   - Audio playback stops on filtered song

 - UI update for audio upload processing on Library page, play button and song length table fields

Note:
 - Does not include the frontend update to load play button and song Length when song has finished processing on the backend. Requires manual refresh of page.

 - Requires a CORS bypass for audio playback due to current access restrictions, used `CORS Unblock` chrome extension during development

 - Might be library bug or some other issue, when the audio player is created on Library page there is a noticeable lag in dragging/moving the window browser around. Double checked with spotify web player and it has same issue, little bit of research led to disabling hardware acceleration to fix it... otherwise no real performance issue seen

### CODE REVIEW

**Review by escobarjonatan — CHANGES_REQUESTED** [review]

File: src/pages/home/library/Discography.tsx [hunk 1]
```diff
@@ -16,8 +18,23 @@ const Discography: FunctionComponent = () => {
...
+    const newAudioPlayer = videojs(audioElement);
+    setAudioPlayer(newAudioPlayer);
+
+    return () => {
+      if (!newAudioPlayer.isDisposed()) {
+        newAudioPlayer.dispose();
+      }
+    };
+  }, []);
+
```
[the remark this answers is not in the document]
Comment (dmkirshon) [reply src/pages/home/library/Discography.tsx:37]: Strange, ok, I thought I tried using `useRef` and was running into an issue where even with the unmount disposal the player would not be removed and continue playing on other pages. Switched to useState to keep it within the component/page.

Is there a reason for having both `useState` and `useRef`? Should I use the `audioPlayerRef` for the pause, src, and play functionalities below that?

What do you mean by ensure access to the DOM element? Like for future visual audio player use/manipulation on other pages and you just meant a reference to the audio element created? Just from a javascript side of things, from what I understand the `new Audio()` works in the background as an in-memory javascript audio element, and doesn't actually add the audio element to the DOM, just technicalities but wanted to clarify.

File: src/pages/home/library/SongList.tsx [hunk 2]
```diff
@@ -173,16 +174,25 @@ export default function SongList({
...
+                              "Upload in progress. Please allow a few " +
+                              "minutes to complete the process."
+                            }
+                          >
+                            <CircularProgress
+                              disableShrink
+                              size={ 20 }
+                              color="secondary"
+                            />
+                          </Tooltip>
```
[the remark this answers is not in the document]
Comment (dmkirshon) [reply src/pages/home/library/SongList.tsx:195]: No problem was looking at all of the `SongList` page and going _man this should get refactored somehow..._ might need to revisit the whole file to separate and clean up the different features added post MVP. Unless it's seen as a worthy endeavor right now, let me know. For now will work off your suggestion Jonatan. Thank you!

File: src/pages/home/library/Discography.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (dmkirshon) [reply src/pages/home/library/Discography.tsx:37]: Also, do you think I should create an `audioPlayer` util and refactor this out of `Discography`? It's simple enough now but I'm sure this is going to become a more robust feature, even if it is just for Artist previewing their own songs right now.

File: src/pages/home/library/SongList.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (dmkirshon) [reply src/pages/home/library/SongList.tsx:195]: Looking through,I assume the path without refactoring everything would be converting the 'songData.map' from an implicit return into one with '{}' and a return statement?

File: src/pages/home/library/Discography.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (escobarjonatan) [reply src/pages/home/library/Discography.tsx:37]: >Is there a reason for having both useState and useRef? Should I use the audioPlayerRef for the pause, src, and play functionalities below that?
You could, i didnt test that locally 

State is usually used to hold a piece of information and ref is used to hold a reference to an actual element. Reading it again it does sound a bit confusing, sorry.
"The Audio() constructor creates and returns a new HTMLAudioElement which can be either attached to a document for the user to interact with and/or listen to, or can be used offscreen to manage and play audio" - [MDN](https://developer.mozilla.org/en-US/docs/Web/API/HTMLAudioElement/Audio).

>do you think I should create an audioPlayer util and refactor this out of Discography
Not sure juice is worth the squeeze here imo, i think it's okay to keep it here

File: src/pages/home/library/SongList.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (escobarjonatan) [reply src/pages/home/library/SongList.tsx:195]: I really like that last comment, if it's not too complicated to do, i would move it to its own component. If not, the variables will suffice

File: src/pages/home/library/Discography.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (dmkirshon) [reply src/pages/home/library/Discography.tsx:49]: Makes sense, updated with useRef to handle the audio player element

File: src/pages/home/library/SongList.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (dmkirshon) [reply src/pages/home/library/SongList.tsx]: Updated with new `SongStreamPlaybackIcon` component

**Review by escobarjonatan — APPROVED** [review]

**Pull request merged**